### ISSUE TITLE
Migrate on pytest

### ISSUE THREAD

**Issue by orsinium** [life4/deal#16]
Simple and important task. We have a bunch of unittest-based tests that are difficult to maintain and write new. We're already running them with pytest. The next step is change unittest-based test on pytest-based. For example:

Old:

```python
class PreTest(unittest.TestCase):
    def test_main(self):
        func = deal.pre(lambda x: x > 0)(lambda x: x)

        with self.subTest(text='good'):
            self.assertEqual(func(4), 4)

        with self.subTest(text='error'):
            with pytest.raises(deal.PreContractError):
                func(-2)
```

New:

```python
(in tests/test_pre.py)
def test_main():
        func = deal.pre(lambda x: x > 0)(lambda x: x)
        assert  func(4) == 4
        with pytest.raises(deal.PreContractError):
                func(-2)
```

Use power of `pytest.mark.parametrize`, be creative :fire:

**Comment by OsamaNabih** [comment]
Must I be familiar with unittest and pytest to start working on this?

**Comment by orsinium** [comment]
Yes, but they are reeeeealy simple and useful. It's worth it. Good luck,

**Comment by OsamaNabih** [comment]
I'll try to get started on it. I'm not sure if I'll be able to complete it so anyone else is free to work on it too. :)

**Comment by orsinium** [comment]
Closed in #19
